### ISSUE TITLE
Feature request: Everything search integration

### ISSUE THREAD

**Issue by yaichenbaum** [files-community/Files#5845]
The current solution for searching for files does work, but I believe we can add value if we use Everything Search when we detect it's installed. Everything Search can index and search for files much faster than the current method and will help users be more productive.

**Comment by Vulcanraven91** [comment]
I agree with you, I'm using it like 5 years now. Great tool.
Here is a SDK Wrapper in Fluent.

https://github.com/Riboe/EverythingSharp

**Comment by Vulcanraven91** [comment]
I agree with you, I'm using it like 5 years now. Great tool.
Here is a SDK Wrapper in Fluent.

https://github.com/Riboe/EverythingSharp

**Comment by yaichenbaum** [comment]
Does the wrapper require users to have Everything installed?

**Comment by heftymouse** [comment]
It does, since Everything's SDK operates through IPC with the running Everything instance

**Comment by yaichenbaum** [comment]
We'll have to find an alternative

**Comment by RedIODev** [comment]
Why require to not need extra steps?
I mean don't get me wrong less steps is always better.
But in this case we are talking about the integration of an optional alternative search engine.
It would be way more flexible and easier to implement to just make a pair of radio buttons in the settings:
[x] System-Search
[ ] Everything-Search

where Everything-Search is grayed out if not running on the system and add a hyperlink to the voidtools website.
This would also enable the possibility to support multiple optional search engines when implementing this system in a modular way. 
Of cause another alternative would be to bundle Everything. But for now ignoring legal stuff this would bloat Files for people who for some reason want to use the windows search and would be really fragile being directly dependent on some foreign search engine still running separately on the system but this time started by Files. A search indexer needs to be it's own programm since it needs to sun continuously in the background and can therefore never be directly integrated in files. Just like the current search is also hosted by windows also running in an separate program. (Assuming this is how the search works. Which I didn't check) 
Another point for the soft/dynamic integration of Everything is that other software that uses the Everything Api can share the same Everything instance with Files. Not needing duplicate instances and installations.

**Comment by yaira2** [comment]
> It would be way more flexible and easier to implement to just make a pair of radio buttons in the settings:

Adding an integration can sometimes take as much effort as building the feature from scratch.

**Comment by RedIODev** [comment]
I currently don't have the time. But when this thread is still open in a few weeks (likely) I will try implementing it. I'm not familiar with the Files source but in C# it shouldn't be to difficult since there already is an c# api like mentioned by 

# # # @ @

> I agree with you, I'm using it like 5 years now. Great tool. Here is a SDK Wrapper in Fluent.
> 
> https://github.com/Riboe/EverythingSharp

I didn't look into it throughly but it seems most of the work is already done. Only thing needed is the ui integration and rewiring of the search box.

**Comment by yaira2** [comment]
Please let me know if you plan to work on improving search so that we can collaborate. Everything Search integration is possible but only if it's bundled with the app.

**Comment by Svk1190** [comment]
Here is a related project - [Everything Toolbar](https://github.com/stnkl/EverythingToolbar). It might be nice to have such a search bar in Files.

**Comment by Johndo1234** [comment]
+1
I think it would be very useful if the files search bar was replaced with everything search.

**Comment by Fefedu973** [comment]
+1 very useful

**Comment by Fefedu973** [comment]
Instead of replacing all the search bars (could be optional) I think it would be better to add a "main" search bar on the home page to search everything in the computer only using everything search

**Comment by Surferlul** [comment]
You can use everything search to search subdirectories, so I'd find it quite useful to have everywhere. Otherwise you'll have slow search for subdirectories... Even if you'd want custom filtering on subdirectory searches, you could use everything to give you all files in the directory and do custom filtering on that information. This would make files the only file manager (that I know of) with universal instant search which is a huge plus

**Comment by Fefedu973** [comment]
Yes we could use everything search everywhere with filters too (coming soon I think) but a main search bar to search all the computer really belong to the home page in NY opinion it would fit in really well

**Comment by Welding-Torch** [comment]
+1 much needed

**Comment by jason919** [comment]
whatever you guys decide, please do something to current search, it is way too slow.

**Comment by ndekross** [comment]
Yes, please.

**Comment by GeoDoX** [comment]
What's the status of this issue? I'm seeing other issues closed as completed. Has Everything been integrated? Can this be closed?

**Comment by Josh65-2201** [comment]
The other were closed as duplicate of this one. A way of integrating it following the requirements at the top hasn't been found yet.

**Comment by jeregon1** [comment]
So, it's not possible using the library https://github.com/Riboe/EverythingSharp ?

**Comment by Josh65-2201** [comment]
That requires everything to be installed and running in the background. It doesn't meet the requirements.

**Comment by yaira2** [comment]
There is a way to bundle the dlls together with Files, but it requires further investigation. I recommend trying our Listary integration in the meantime.

**Comment by sky96111** [comment]
> ### Steps to implement
> * Integrate Everything Search with Files. The integration shouldn't require any extra steps on the users side (eg. installing Everything).

Honestly, I don't think it is not a good idea to bundle Everything to Files App.

I prefer this intergration.
> But in this case we are talking about the integration of an optional alternative search engine.
> It would be way more flexible and easier to implement to just make a pair of radio buttons in the settings:
> [x] System-Search
> [ ] Everything-Search

Everything needs to run in the background or service to get the ability to automatically update NTFS volume index. However bundling will limit volume update task to Files App lifecicle. And running multiple Everything instances in the background.

Everything provides an SDK for third-party applications to use IPC calls. (e.g. utools, EverythingPowerToys) Which can reuse Everything instance.

**Comment by jeregon1** [comment]
Honestly, I think it's the best, just as Listary is not bundled nor Windows Indexer, I think it would be a flexible option to have Everything as an option.

**Comment by heide-oficial** [comment]
2025 and still no integration with Everything 😟

**Comment by ARHAEEM** [comment]
finally in progress! 🎉

**Comment by Omena0** [comment]
This should also be used for **folder sizes**, if indexing them is enabled in Everything, of course.

**Comment by Kacper-Kondracki** [comment]
There is Windhawk Windows Explorer mod that integrates folder sizes with Everything. It is blazingly fast 🔥. Folder size integration in Files is desirable.

**Comment by gilnett** [comment]
i think they know it but files is different from default explorer and don't work the same way

**Comment by DayDreamerDD** [comment]
So...as far as I saw in full changelog, this pr was not merged into main branch. It's still under development or abandoned?
#17336

**Comment by gilnett** [comment]
I don't know but i think it is not planned for now

**Comment by Josh65-2201** [comment]
The PR was sent to be continued by some else. there is no update on their progress yet

**Comment by yair100** [comment]
It's available if anyone wants to work on it, but please reach out beforehand so we can discuss the relevant details. #17336 might be a good starting point for anyone working on this issue, but it's a bit rough and doesn't meet our goals for how the integration should work.

**Comment by Vocllum** [comment]
There's a "Better file sizes in Explorer details" mod in Windhawk which can mod Windows. The mod can make it possible for explorer to load folder size via Everything service. It could be vary handful. Displaying folder sizes in Files is currently too performance-intensive, and this would be an extremely good solution.
<img width="2560" height="1416" alt="Image" src="https://github.com/user-attachments/assets/c6632f17-78df-4237-ab3a-4176f3859b37" />

**Comment by Wingingbump** [comment]
I'd like to give my try at this. Are we still trying to integrate Everything search at this point? I've created a fork that adds a separate Rust process that maintains its own Tantivy filename index. Didn't see this issue before I started though.

**Comment by Josh65-2201** [comment]
How to do it still in discussion, #17336 is a good start point

**Comment by yair100** [comment]
@Wingingbump I would prefer that we maintain our own index. Would you like to join our Discord server to discuss this in more detail?

**Comment by Wingingbump** [comment]
Okay, reached out in #general in discord. I've also integrated Everything Search as an add-on feature in this branch but it depends on using their indexer. https://github.com/Wingingbump/Files/tree/feature/everything-search

**Comment by yair100** [comment]
Thanks, we'll continue in Discord.

**Comment by Omena0** [comment]
Why maintain your own index? Is there some specific data you need that Everything doesn't provide (by default?)? If not then I don't see why you'd want a separate file scanner running in the background and a separate index using storage space. Its not like Everything would have to be a dependency, yeah sure you can add a fallback indexer or not and use Everything if its available.

**Comment by yair100** [comment]
Our goal is to have a fast search experience for everyone without requiring users to download a sperate program. Additionally, having our own index will allow us to speed up other aspects of the app as well.
